highfidelity/hifi
# Soft Attachment Support

:warning: AvatarMixer protocol change.

Items in the attachment menu "Edit > Attachments..." now have the isSoft flag.  Soft attachments with vertex weights will animate with the base avatar model, this feature can be used to attach clothing to an avatar.

To Test:
1. In preferences dialog change the avatar url to the following: https://hifi-content.s3.amazonaws.com/ozan/dev/avatars/hifi_default_avatar/hifi_default_body_modular.fst
2. Open the edit attachments dialog, add this coat model as a 'soft' attachment: https://hifi-content.s3.amazonaws.com/ozan/dev/clothes/coat/coat_rig.fbx
3. Also add a 'rigid' attachment to the "Head" (not "head") joint: https://s3.amazonaws.com/hifi-public/tony/cowboy-hat.fbx
4. Your avatar should now be wearing a coat and a hat.

Due to the protocol change you must be running an AvatarMixer and a client build from this branch to see the changes reflect over the network.
